File: patient_app/lib/data/models/doctor_model.dart
// lib/features/doctor/data/models/doctor_model.dart
class Doctor {
  final String id;
  final String name;
  final String specialization;
  final String qualification;
  final int experience;
  final double rating;
  final int reviewCount;
  final String? profileImage;
  final String clinicName;
  final String clinicAddress;
  final double? distance;
  final double consultationFee;
  final bool isVerified;

  Doctor({
    required this.id,
    required this.name,
    required this.specialization,
    required this.qualification,
    required this.experience,
    required this.rating,
    required this.reviewCount,
    this.profileImage,
    required this.clinicName,
    required this.clinicAddress,
    this.distance,
    required this.consultationFee,
    required this.isVerified,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      specialization: json['specialization'] ?? '',
      qualification: json['qualification'] ?? '',
      experience: json['experience'] ?? 0,
      rating: (json['rating'] ?? 0).toDouble(),
      reviewCount: json['reviewCount'] ?? 0,
      profileImage: json['profileImage'],
      clinicName: json['clinicName'] ?? '',
      clinicAddress: json['clinicAddress'] ?? '',
      distance: json['distance'] != null ? (json['distance'] as num).toDouble() : null,
      consultationFee: (json['consultationFee'] ?? 0).toDouble(),
      isVerified: json['isVerified'] ?? false,
    );
  }
}

File: backend/src/doctor-channeling/models/doctor.model.js
const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema({
    name: { type: String, required: true },
    specialization: { type: String, required: true },
    qualification: { type: String, default: '' },
    experience: { type: Number, default: 0 },
    rating: { type: Number, default: 0, min: 0, max: 5 },
    reviewCount: { type: Number, default: 0 },
    profileImage: { type: String, default: null },
    clinicName: { type: String, default: '' },
    clinicAddress: { type: String, default: '' },
    consultationFee: { type: Number, default: 0 },
    clinicId: { type: mongoose.Schema.Types.ObjectId, ref: 'ClinicLocation' },
    city: { type: String },
    district: { type: String },
    contactNumber: { type: String },
    isVerified: { type: Boolean, default: false },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
});

doctorSchema.index({ name: 'text', specialization: 'text', clinicName: 'text' });

module.exports = mongoose.model('Doctor', doctorSchema);
